src/Adapter/ContextStateManager.php

```diff
@@ -154,6 +157,25 @@ public function setCustomer(?Customer $customer): self
         return $this;
     }
 
+    /**
+     * Sets context shop and saves previous value
+     *
+     * @param Shop|null $shop
+     * @param int $shopContextType
+     *
+     * @return $this
+     *
+     * @throws \PrestaShopException
+     */
+    public function setShop(?Shop $shop, int $shopContextType = Shop::CONTEXT_SHOP): self
+    {
+        $this->saveContextField('shop');
+        $this->context->shop = $shop;
+        Shop::setContext(Shop::CONTEXT_SHOP, null !== $shop ? $shop->id : null);
```

Comment:
I hesitate to for the ALL_SHOP shop context when `$shop` is null?

Reply:
I hesitate too 😛 I am not 100% sure that `$shop === null` is equivalent to ALL_SHOP ... 😅 for example what about CLI context and Webservice context

Reply:
Yep that's why for now I thought it may be less risky to force single shop context when this setter is used but not assume any other behaviour when null is used 🤷‍♂️
Or maybe we should even not accept null and it would solve the problem